statsmodels/survey/summary_stats.py

```diff
@@ -0,0 +1,787 @@
+        if self.design.rep_weights is None:
+            nh = self.design.clust_per_strat[self.design.strat_for_clust].astype(np.float64)
+            _pseudo = jdata + nh[:, None] * (np.dot(self.design.weights,
+                                                    self.data) - jdata)
+            mh = np.sqrt((nh - 1) / nh)
+            fh = np.sqrt(1 - self.design.fpc)
+            jdata = fh[:, None] * mh[:, None] * jdata
+        else:
+            nh = self.design.rep_weights.shape[1]
+            mh = (nh - 1) / nh
+            jdata *= np.sqrt(mh)
+        self.vcov = np.dot(jdata.T, jdata)
+        if self.vcov.ndim == 2:
+            self.stderr = np.sqrt(np.diag(self.vcov))
+        else:
+            self.stderr = np.sqrt(self.vcov)
+
+    def _linearized(self):
+        if self.center_by != 'stratum':
+            raise ValueError("Must center by stratum with linearized variance")
+        self.est = self._stat(self.design.weights)
+        jdata = []
+        # for each cluster
+        for c in range(self.design.n_clust):
+            w = self.design.weights.copy()
+            # but if you're not in that cluster, set as 0
+            w[self.design.clust != c] = 0
+            jdata.append(self._weighted_score(w))
+        jdata = np.asarray(jdata)
+        # we usually deal w/ jdata as nxp
+        # unless w/ ratio, in which 2 columns
+        if jdata.ndim == 1:
+            jdata = jdata[:, None]
+        for s in range(self.design.n_strat):
+            # center the 'delete 1' statistic
+            jdata[self.design.ii[s], :] -= jdata[self.design.ii[s],
+                                                 :].mean(0)
+        nh = self.design.clust_per_strat[self.design.strat_for_clust].astype(np.float64)
+        mh = np.sqrt(nh / (nh-1))
+        fh = np.sqrt(1 - self.design.fpc)
+        jdata = fh[:, None] * mh[:, None] * jdata
+        self.vcov = np.dot(jdata.T, jdata)
+
+        if self.vcov.ndim == 2:
+            self.stderr = np.sqrt(np.diag(self.vcov))
+        else:
+            self.stderr = np.sqrt(self.vcov)
+
+
+class SurveyMean(SurveyStat):
+    """
+    Calculates the mean for each column.
+
+    Parameters
+    -------
+    design : SurveyDesign object
+    data : ndarray
+        nxp array of the data to calculate the mean on
+    method: string
+        User inputs whether to get bootstrap or jackknife SE
+
+    Attributes
+    ----------
+    data : ndarray
+        The data which to calculate the mean on
+    design :
+        Points to the SurveyDesign object
+    est : ndarray
+        The point estimates of the statistic, calculated on the columns
+        of data.
+    vcov : ndarray
+        The variance-covariance of the estimates.
+    pseudo : ndarray
+        The jackknife pseudo-values.
+    """
+    def __init__(self, design, data, cov_method='jack', center_by='global',
+                 n_reps=None, bsn=None):
+        super().__init__(design, data, center_by)
```

Comment:
@josef-pkt this raises an lgtm error but we're not sure how to fix it.  lgtm thinks we are not calling the superclass initializer, but we seem to be doing so, via super.

Reply:
lgtm has some false positives, there are many in discrete because, for example, we intentionally skip a class in super.

`super()`  needs the arguments `(class, self)` at least for older python (and for explicitness)